src/frontend/sharedComponents/PermissionAudio.tsx

```diff
@@ -1,72 +1,100 @@
   isOpen: boolean;
+  sheetRef: React.RefObject<BottomSheetModalMethods>;
 }
 
-export const PermissionAudio: FC<PermissionAudio> = props => {
-  const {sheetRef, closeSheet, isOpen} = props;
+export const PermissionAudio: FC<PermissionAudioProps> = ({
+  closeSheet,
+  isOpen,
+  sheetRef,
+}) => {
   const {formatMessage: t} = useIntl();
-  const [permissionResponse] = Audio.usePermissions({request: false});
+  const navigation = useNavigationFromRoot();
+  const [permissionResponse, setPermissionResponse] =
+    useState<PermissionResponse | null>(null);
 
-  const handlePermissionGranted = useCallback(() => {
+  const handleOpenSettings = () => {
+    Linking.openSettings();
     closeSheet();
-  }, [closeSheet]);
-
-  const isPermissionGranted = Boolean(permissionResponse?.granted);
+  };
 
-  useEffect(() => {
-    if (isPermissionGranted) handlePermissionGranted();
-  }, [isPermissionGranted, handlePermissionGranted]);
+  const handleRequestPermission = async () => {
+    const response = await Audio.requestPermissionsAsync();
+    setPermissionResponse(response);
+    if (response.status === 'granted') {
+      closeSheet();
+      navigation.navigate('Audio');
+    } else if (response.status === 'denied' && response.canAskAgain) {
+      closeSheet();
+    } else if (response.status === 'denied' && !response.canAskAgain) {
+      handleOpenSettings();
+    }
+  };
 
   let onPressActionButton: () => void;
   let actionButtonText: string;
-  switch (permissionResponse?.status) {
-    case undefined:
-    case PermissionStatus.UNDETERMINED:
-      onPressActionButton = handleOpenSettings;
-      actionButtonText = t(m.allowButtonText);
-      break;
-    case PermissionStatus.DENIED:
-      if (permissionResponse.canAskAgain) {
-        onPressActionButton = handleOpenSettings;
-        actionButtonText = t(m.allowButtonText);
-      } else {
-        onPressActionButton = handleRequestPermissions;
-        actionButtonText = t(m.goToSettingsButtonText);
-      }
-      break;
-    case PermissionStatus.GRANTED:
-      onPressActionButton = handlePermissionGranted;
-      actionButtonText = t(m.allowButtonText);
-      break;
-    default:
-      throw new Error('Unexpected permission response');
+
+  if (!permissionResponse) {
+    onPressActionButton = async () => {
+      await handleRequestPermission();
+    };
+    actionButtonText = t(m.allowButtonText);
+  } else if (permissionResponse.status === 'denied') {
+    onPressActionButton = handleOpenSettings;
+    actionButtonText = t(m.goToSettingsButtonText);
+  } else {
+    onPressActionButton = async () => {
+      await handleRequestPermission();
+    };
+    actionButtonText = t(m.allowButtonText);
   }
 
   return (
     <BottomSheetModal
-      fullScreen
       ref={sheetRef}
+      isOpen={isOpen}
       onDismiss={closeSheet}
-      isOpen={isOpen}>
+      fullScreen>
       <BottomSheetModalContent
         icon={<AudioPermission />}
         title={t(m.title)}
```

Comment:
this should have a [paddingTop of 80](https://www.notion.so/digidem/Spacing-74d09a0bdefa4bc5b9508336f7b2142a?pvs=4#059606ae2b9840018976a1d1400f1836)